sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Statistics.scala

```diff
@@ -184,6 +196,26 @@ object ColumnStat extends Logging {
   }
 
   /**
+   * Converts from string representation of external data type to the corresponding Catalyst data
+   * type.
+   */
+  private def fromExternalString(s: String, name: String, dataType: DataType): Any = {
+    dataType match {
+      case BooleanType => s.toBoolean
+      case _: IntegralType => s.toLong
+      case DateType => DateTimeUtils.fromJavaDate(java.sql.Date.valueOf(s)).toLong
```

Comment:
according to the doc https://github.com/apache/spark/pull/17630/files#diff-a4113ed6a89e8d19a39f5c27ce95658bR79 , this should be int